gammapy/stats/feldman_cousins.py

```diff
@@ -0,0 +1,442 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Provide a Feldman Cousins algorithm."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+from astropy.extern.six.moves import range
+from astropy.extern.six import iteritems
+
+import logging
+log = logging.getLogger(__name__)
+
+__all__ = ['fc_find_acceptance_region_gauss',
+           'fc_find_acceptance_region_poisson',
+           'fc_construct_acceptance_intervals_pdfs',
+           'fc_get_upper_and_lower_limit',
+           'fc_fix_upper_and_lower_limit',
+           'fc_find_limit',
+           'fc_find_average_upper_limit',
+           'fc_construct_acceptance_intervals',
+          ]
+
+
+def fc_find_acceptance_region_gauss(mu, sigma, x_bins, alpha):
+    """Analytically find the acceptance region (x_min, x_max) for Gaussian with
+       boundary at the origin, such that int_x_min^x_max P(x|mu)dx = alpha
+
+    Parameters
+    ----------
+    mu : double
+        Mean of the Gaussian
+    sigma : double
+        Width of the Gaussian
+    x_bins : array-like
+        Bins in x
+    alpha : double
+        Desired confidence level
+
+    Returns
+    -------
+    (x_min, x_max) : tuple of floats
+        Acceptance interval
+    """
+
+    from scipy import stats
+
+    dist = stats.norm(loc=mu, scale=sigma)
+
+    x_bin_width = x_bins[1] - x_bins[0]
+
+    p = []
+    r = []
+
+    for x in x_bins:
+        p.append(dist.pdf(x)*x_bin_width)
+        # This is the formula from the FC paper
+        if mu == 0 and sigma == 1:
+            if x < 0:
+                r.append(np.exp(mu*(x-mu*0.5)))
+            else:
+                r.append(np.exp(-0.5*np.power((x-mu),2)))
+        # This is the more general formula
+        else:
+            # Implementing the boundary condition at zero
+            muBest     = max(0, x)
+            probMuBest = stats.norm.pdf(x, loc=muBest, scale=sigma)
+            if probMuBest == 0.0:
+                r.append(0.0);
```

Comment:
Well, I get the results from the FC paper, so my code can't be completely wrong. I'm also consistent with the ROOT TFeldmanCousins results (that is if I ignore the problems in their implementation).

I think this line is actually not needed because P(n|mu_best) should never be zero. But it doesn't hurt to check for zero before division (the ROOT code does the same test).